modules/notify/main/index.coffee

```diff
@@ -1,33 +1,65 @@
 # utility method for setting up notifications
 setupNotification = (notification, selection) ->
-  if notification.options.icon? and notification.options.icon.length > 0
-    selection.append('div')
-      .class('hx-notification-icon-container')
-      .append('i')
-        .class('hx-notification-icon ' + notification.options.icon)
-
-  selection
-    .append('div')
-      .class('hx-notification-text')
-      .text(notification.message)
+  icon = if notification.options.icon? and notification.options.icon.length > 0
+    hx.detached('div').class('hx-notification-icon-container')
+      .add(hx.detached('i').class('hx-notification-icon ' + notification.options.icon))
+
+  content = hx.detached('div').class('hx-notification-content')
+  msg = notification.message
+
+  msgIsString = hx.isString(msg) or hx.isNumber(msg)
+
+  msgIsNode = not msgIsString and ((msg instanceof hx.Selection) or (msg instanceof HTMLElement))
+  msgIsArrayOfNodes = msgIsNode or hx.isArray(msg) and msg.every (item) ->
+    (item instanceof hx.Selection) or (item instanceof HTMLElement)
+
+  msgIsObject = not msgIsString and not msgIsNode and not msgIsArrayOfNodes and hx.isObject(msg)
+
+  if msgIsString
+    content.text(msg)
+  if msgIsNode or msgIsArrayOfNodes
+    content.add(msg)
+  else if msgIsObject and notification.options.renderer
+    notification.options.renderer(content.node(), msg)
+  else
+    if msgIsObject
+      hx.consoleWarning(
+        'Notification created using an object with invalid arguments\n',
+        'An object was passed to the notification without a renderer being defined\n'
+        "message:",
+        msg,
+        "\nrenderer:",
+        notification.options.renderer
+      )
+    else
+      hx.consoleWarning(
+        'Notification created using an object with invalid arguments\n',
+        'The notification expected a String, Selection, HTMLElement or an Object with matching renderer but was passed:\n',
+        "message:",
+        msg
+      )
+    content.text('ERROR CONSTRUCTING NOTIFICATION')
```

Comment:
I'm not sure how best to handle the error case with the notification construction.
If an error is logged, we don't populate the notification and it then has no content. I thought doing `selection.display('none')` would be reasonable enough but ended up having this instead as it makes it more obvious that something has gone wrong and that there _should_ have been a notification.